src/components/BarChart/BarChart.jsx

```diff
@@ -335,6 +358,46 @@ const BarChart = createClass({
 			? yAxisTooltipDataFormatter
 			: yAxisFinalFormatter;
 
+		if (_.isEmpty(data) || width < 1 || height < 1 || isLoading) {
+			const emptyMessageBodyProp = _.get(getFirst(this.props, BarChart.EmptyMessageBody), 'props');
+			const emptyMessageTitleProp = _.get(getFirst(this.props, BarChart.EmptyMessageTitle), 'props', {children: 'You have no data.'});
+
+			return (
+				<EmptyStateWrapper
+					isEmpty={_.isEmpty(data)}
+					isLoading={isLoading}
+				>
+					<EmptyStateWrapper.EmptyMessageBody {...emptyMessageBodyProp}/>
+					<EmptyStateWrapper.EmptyMessageTitle {...emptyMessageTitleProp}/>
+					<svg
+						{...omitProps(passThroughs, BarChart)}
```

Comment:
there are other components that have this conflict between root passThrus vs expectec passThrus, and so far, we kept them at the root level only, and created props that pass thru to the underlying elements as needed. we should keep with that pattern until we reach a major version upgrade.